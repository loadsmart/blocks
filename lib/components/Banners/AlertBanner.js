"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = __importStar(require("react"));
var react_native_1 = require("react-native");
var index_1 = require("../../res/index");
var BannerProps_1 = require("./BannerProps");
var AlertBanner = /** @class */ (function (_super) {
    __extends(AlertBanner, _super);
    function AlertBanner() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    AlertBanner.prototype.render = function () {
        var styles = stylesFromProps(this.props);
        return (react_1.default.createElement(react_native_1.TouchableWithoutFeedback, { onPress: this.props.onPress },
            react_1.default.createElement(react_native_1.View, { style: [styles.wrapper, this.props.style] },
                react_1.default.createElement(react_native_1.Image, { source: index_1.Images.Warning }),
                this.props.message && (react_1.default.createElement(react_native_1.Text, { style: styles.text, numberOfLines: 2 }, this.props.message)))));
    };
    return AlertBanner;
}(react_1.PureComponent));
exports.default = AlertBanner;
var stylesFromProps = function (props) {
    return react_native_1.StyleSheet.create({
        wrapper: {
            flexDirection: 'row',
            height: 55,
            alignItems: 'center',
            backgroundColor: props.displayStyle === BannerProps_1.BannerDisplayStyle.Error ? index_1.Colors.CoralPink : index_1.Colors.OrangePeel,
            paddingHorizontal: 15,
        },
        text: {
            fontFamily: index_1.Fonts.SharpSansSemibold,
            fontSize: 14,
            color: index_1.Colors.White,
            marginLeft: 12,
        },
    });
};
