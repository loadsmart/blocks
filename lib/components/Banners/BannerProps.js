"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var BannerDisplayStyle;
(function (BannerDisplayStyle) {
    BannerDisplayStyle[BannerDisplayStyle["Warning"] = 0] = "Warning";
    BannerDisplayStyle[BannerDisplayStyle["Error"] = 1] = "Error";
})(BannerDisplayStyle = exports.BannerDisplayStyle || (exports.BannerDisplayStyle = {}));
