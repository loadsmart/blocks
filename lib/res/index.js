"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var colors_1 = __importDefault(require("./colors"));
exports.Colors = colors_1.default;
var fonts_1 = __importDefault(require("./fonts"));
exports.Fonts = fonts_1.default;
var images_1 = __importDefault(require("./images"));
exports.Images = images_1.default;
