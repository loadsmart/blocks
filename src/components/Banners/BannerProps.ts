import { ImageSourcePropType, ViewProps } from 'react-native'

export enum BannerDisplayStyle {
  Warning,
  Error,
}

export interface BannerProps extends ViewProps {
  displayStyle?: ImageSourcePropType
  message: string
  onPress?: () => void
}
