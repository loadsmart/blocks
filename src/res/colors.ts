const Colors = {
  AlgaeGreen: '#14d64d',
  BrownGreyThree: '#9b9b9b',
  Charcoal: '#3c3d3a',
  CoralPink: '#fd576a',
  DarkGrey: '#666666',
  OrangePeel: '#ff9800',
  LightGray: '#d4d4d4',
  Tuna: '#484a4c',
  VeryLightGray: '#eee',
  White: '#fff',
  Ghost: '#c4ccd6',
  Transparent: '#00000000',
}

export default Colors
