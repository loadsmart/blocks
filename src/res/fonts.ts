const Fonts = {
  SharpSansBold: 'SharpSans-Bold',
  SharpSansItalic: 'SharpSans-BoldItalic',
  SharpSansBook: 'SharpSans-Book',
  SharpSansBookItalic: 'SharpSans-BookItalic',
  SharpSansExtrabold: 'SharpSans-Extrabold',
  SharpSansExtraboldItalic: 'SharpSans-ExtraboldItalic',
  SharpSansLight: 'SharpSans-Light',
  SharpSansLightItalic: 'SharpSans-LightItalic',
  SharpSansMedium: 'SharpSans-Medium',
  SharpSansMediumItalic: 'SharpSans-MediumItalic',
  SharpSansSemibold: 'SharpSans-Semibold',
  SharpSansSemiboldItalic: 'SharpSans-SemiboldItalic',
  SharpSansThin: 'SharpSans-Thin',
  SharpSansThinItalic: 'SharpSans-ThinItalic',
}

export default Fonts
