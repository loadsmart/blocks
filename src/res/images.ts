const Images = {
  Chevron: require('../../assets/images/chevron.png'),
  PlusSign: require(`../../assets/images/plus-sign.png`),
  SmallCheck: require(`../../assets/images/small-check.png`),
  Warning: require(`../../assets/images/warning.png`),
  PasswordShow: require('../../assets/images/password-show.png'),
  PasswordHide: require('../../assets/images/password-hide.png'),
}

export default Images
