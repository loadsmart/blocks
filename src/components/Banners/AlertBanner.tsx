import React, { PureComponent } from 'react'
import { Image, StyleSheet, Text, TouchableWithoutFeedback, View } from 'react-native'
import { Colors, Fonts, Images } from '../../res/index'
import { BannerDisplayStyle, BannerProps } from './BannerProps'

export default class AlertBanner extends PureComponent<BannerProps> {
  public render() {
    const styles = stylesFromProps(this.props)
    return (
      <TouchableWithoutFeedback onPress={this.props.onPress}>
        <View style={[styles.wrapper, this.props.style]}>
          <Image source={Images.Warning} />
          {this.props.message && (
            <Text style={styles.text} numberOfLines={2}>
              {this.props.message}
            </Text>
          )}
        </View>
      </TouchableWithoutFeedback>
    )
  }
}

const stylesFromProps = (props: BannerProps) => {
  return StyleSheet.create({
    wrapper: {
      flexDirection: 'row',
      height: 55,
      alignItems: 'center',
      backgroundColor:
        props.displayStyle === BannerDisplayStyle.Error ? Colors.CoralPink : Colors.OrangePeel,
      paddingHorizontal: 15,
    },
    text: {
      fontFamily: Fonts.SharpSansSemibold,
      fontSize: 14,
      color: Colors.White,
      marginLeft: 12,
    },
  })
}
