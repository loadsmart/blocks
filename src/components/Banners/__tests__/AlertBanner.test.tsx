import React from 'react'
import renderer from 'react-test-renderer'
import AlertBanner from '../AlertBanner'
import { Image, Text, TouchableWithoutFeedback } from 'react-native'
import { Images } from '../../../res/'
import { BannerDisplayStyle } from '../BannerProps'

describe('AlertBanner', () => {
  it('has an icon', () => {
    const testRenderer = renderer.create(<AlertBanner message={'hello'} />)
    const icon = testRenderer.root.findByType(Image)
    expect(icon.props.source).toBe(Images.Warning)
  })

  it('has the title I passed', () => {
    const testRenderer = renderer.create(<AlertBanner message={'hello'} />)
    const title = testRenderer.root.findByType(Text)
    expect(title.props.children).toBe('hello')
  })

  it('renders for warning style', () => {
    const testRenderer = renderer.create(
      <AlertBanner message={'hello'} displayStyle={BannerDisplayStyle.Warning} />
    )
    expect(testRenderer.toJSON()).toMatchSnapshot()
  })

  it('renders for error style', () => {
    const testRenderer = renderer.create(
      <AlertBanner message={'hello'} displayStyle={BannerDisplayStyle.Error} />
    )
    expect(testRenderer.toJSON()).toMatchSnapshot()
  })

  it('executes onPress callback', () => {
    const onPressCallback = jest.fn()
    const testRenderer = renderer.create(
      <AlertBanner message={'hello'} onPress={onPressCallback} />
    )

    const touchable = testRenderer.root.findByType(TouchableWithoutFeedback)
    touchable.props.onPress()

    expect(onPressCallback).toBeCalledTimes(1)
  })
})
