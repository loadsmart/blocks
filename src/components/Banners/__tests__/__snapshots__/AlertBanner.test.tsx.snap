// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertBanner renders for error style 1`] = `
<View
  accessible={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#fd576a",
        "flexDirection": "row",
        "height": 55,
        "paddingHorizontal": 15,
      },
      undefined,
    ]
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../assets/images/warning.png",
      }
    }
  />
  <Text
    numberOfLines={2}
    style={
      Object {
        "color": "#fff",
        "fontFamily": "SharpSans-Semibold",
        "fontSize": 14,
        "marginLeft": 12,
      }
    }
  >
    hello
  </Text>
</View>
`;

exports[`AlertBanner renders for warning style 1`] = `
<View
  accessible={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#ff9800",
        "flexDirection": "row",
        "height": 55,
        "paddingHorizontal": 15,
      },
      undefined,
    ]
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../assets/images/warning.png",
      }
    }
  />
  <Text
    numberOfLines={2}
    style={
      Object {
        "color": "#fff",
        "fontFamily": "SharpSans-Semibold",
        "fontSize": 14,
        "marginLeft": 12,
      }
    }
  >
    hello
  </Text>
</View>
`;
